client/src/api/block_builder/input_selection/mod.rs

```diff
@@ -352,30 +349,51 @@ pub fn try_select_inputs(
 // Dedup inputs by output id, because other data could be different, even if it's the same output
 fn dedup_inputs(mandatory_inputs: &mut Vec<InputSigningData>, additional_inputs: &mut Vec<InputSigningData>) {
     // Sorting inputs by OutputId so duplicates can be safely removed.
-    mandatory_inputs.sort_by_key(|input| (input.output_metadata.transaction_id, input.output_metadata.output_index));
-    mandatory_inputs.dedup_by_key(|input| (input.output_metadata.transaction_id, input.output_metadata.output_index));
-    additional_inputs.sort_by_key(|input| (input.output_metadata.transaction_id, input.output_metadata.output_index));
-    additional_inputs.dedup_by_key(|input| (input.output_metadata.transaction_id, input.output_metadata.output_index));
+    mandatory_inputs.sort_by_key(|input| {
+        (
+            *input.output_metadata.transaction_id(),
+            input.output_metadata.output_index(),
+        )
+    });
+    mandatory_inputs.dedup_by_key(|input| {
+        (
+            *input.output_metadata.transaction_id(),
+            input.output_metadata.output_index(),
+        )
+    });
+    additional_inputs.sort_by_key(|input| {
+        (
+            *input.output_metadata.transaction_id(),
+            input.output_metadata.output_index(),
+        )
+    });
+    additional_inputs.dedup_by_key(|input| {
+        (
+            *input.output_metadata.transaction_id(),
+            input.output_metadata.output_index(),
+        )
+    });
 
     // Remove additional inputs that are already mandatory.
     // TODO: could be done more efficiently with itertools unique?
     additional_inputs.retain(|input| {
         !mandatory_inputs.iter().any(|mandatory_input| {
-            input.output_metadata.transaction_id == mandatory_input.output_metadata.transaction_id
-                && input.output_metadata.output_index == mandatory_input.output_metadata.output_index
+            input.output_metadata.transaction_id() == mandatory_input.output_metadata.transaction_id()
+                && input.output_metadata.output_index() == mandatory_input.output_metadata.output_index()
         })
     });
 }
 
+// TODO is this still required?
```

Comment:
No can be removed now, it was there because the data from the node could have something which isn't a valid OutputId, but when we know that it's valid because it's required during the creation, then we can remove it